tests/manage/monitoring/conftest.py

```diff
@@ -431,3 +442,421 @@ def corrupt_pg():
+        if line == "{":
+            break
+        else:
+            logger.info(line)
+    fio_parseable_output = "\n".join(fio_output_lines[line_num:])
+    fio_report = yaml.safe_load(fio_parseable_output)
+    return fio_report
+
+
+def workload_fio_storageutilization(
+        fixture_name,
+        target_percentage,
+        project,
+        fio_pvc_dict,
+        fio_job_dict,
+        fio_configmap_dict,
+        measurement_dir,
+        tmp_path,
+):
+    """
+    This function implements core functionality of fio storage utilization
+    workload fixture. This is necessary because we can't parametrize single
+    general fixture over multiple parameters (it would mess with test case id
+    and polarion test case tracking).
+    """
+    if fixture_name.endswith("rbd"):
+        storage_class_name = "ocs-storagecluster-ceph-rbd"
+        ceph_pool_name = "ocs-storagecluster-cephblockpool"
+    elif fixture_name.endswith("cephfs"):
+        storage_class_name = "ocs-storagecluster-cephfs"
+        ceph_pool_name = "ocs-storagecluster-cephfilesystem-data0"
+    else:
+        raise Exception("unexpected volume type, ocs-ci code is wrong")
+
+    # make sure we communicate what is going to happen
+    logger.info((
+        f"starting {fixture_name} fixture, "
+        f"using {storage_class_name} storage class "
+        f"backed by {ceph_pool_name} ceph pool"))
+
+    pvc_size = get_storageutilization_size(target_percentage, ceph_pool_name)
+
+    # For cephfs we can't use fill_fs because of BZ 1763808 (the process
+    # will get *Disk quota exceeded* error instead of *No space left on
+    # device* error).
+    # On the other hand, we can't use size={pvc_size} for rbd, as we can't
+    # write pvc_size bytes to a filesystem on a block device of {pvc_size}
+    # size (obviously, some space is used by filesystem metadata).
+    if fixture_name.endswith("rbd"):
+        fio_conf = textwrap.dedent("""
+            [simple-write]
+            readwrite=write
+            buffered=1
+            blocksize=4k
+            ioengine=libaio
+            directory=/mnt/target
+            fill_fs=1
+            """)
+    else:
+        fio_conf = textwrap.dedent(f"""
+            [simple-write]
+            readwrite=write
+            buffered=1
+            blocksize=4k
+            ioengine=libaio
+            directory=/mnt/target
+            size={pvc_size}G
+            """)
+
+    # put the dicts together into yaml file of the Job
+    fio_configmap_dict["data"]["workload.fio"] = fio_conf
+    fio_pvc_dict["spec"]["storageClassName"] = storage_class_name
+    fio_pvc_dict["spec"]["resources"]["requests"]["storage"] = f"{pvc_size}Gi"
+    fio_objs = [fio_pvc_dict, fio_configmap_dict, fio_job_dict]
+    fio_job_file = ObjectConfFile(fixture_name, fio_objs, project, tmp_path)
+
+    # how long do we let the job running while writing data to the volume
+    write_timeout = pvc_size * 30  # seconds
```

Comment:
why 30 seconds?

Reply:
I need to specify some timeout to prevent `wait_for_resource` method to kill the job before it finishes, and on the other hand, I want this code to fail when the job gets stuck.
The particular value is based solely on my experience with the execution on defaut sized AWS clusters, and I'm willing to change it based on additional data or requirements, or to make it configurable.
Checking screenshot from Fri, I see that I was getting about 0.25s per GB, so there definitely is an opportunity for extending this margin a bit.

Reply:
@red-hat-storage/ocs-qe-ecosystem Do we track expected write rate on given platforms? Here I need to come up with some reasonable timeout wrt number of GB qe code is going to write on cephfs or rbd based volume. Would it make sense to make this configurable, so that CI could tweak this number according to expected performance of the environment being installed?

Reply:
@keesturam do we have any data about the above?

Reply:
Since we got no replies, I'm going to leave this as it is.